src/librustdoc/html/static/main.js

```diff
@@ -360,15 +368,68 @@
                 }
             }
 
+            function extractGenerics(val) {
+                var values = val.split("<")
+                                .slice(1)
+                                .join("")
+                                .split(">")
+                                .slice(0, -1)
+                                .join("")
+                                .split(",");
+                for (var x = 0; x < values.length; ++x) {
+                    values[x] = values[x].trim();
+                }
+                return values;
+            }
```

Comment:
wat.

I would have used `indexOf`/`lastIndexOf` to extract the substring instead of the split.slice.join.split.slice.join

```js
function extractGenerics(val) {
    var values = val.substring(val.indexOf('<')+1, val.lastIndexOf('>'));
    return values.split(/\s*,\s*/);
}
```

Reply:
Way better, thanks!